### DISCUSSION TITLE
配置选项 | OhMyLive2D

### DISCUSSION THREAD

**Discussion by giscus** [oh-my-live2d/oh-my-live2d#34] [Announcements]
# 配置选项 | OhMyLive2D

支持所有Live2D模型版本的Web组件, 快速为您的个人网站添加看板娘.

https://oml2d.com/options/Options.html

**Comment by hacxy** [comment]
好!

**Comment by Yansr82** [comment]
是否有從客戶端自定義的方法?

**Reply by KISGP** [reply]
客户端自定义的方法具体是什么意思🤔?能打个比方吗

**Comment by Yansr82** [comment]
添加一個settings到menu
可以客製顏色之類的

kis ***@***.***> 於 2024年5月2日 週四 下午1:57寫道：

**Reply by KISGP** [reply]
1. 如果你只想更改menu的颜色而不关心其它模块的颜色，可以通过修改 [primaryColor](https://oml2d.com/options/Options.html#primarycolor) 简单的修改颜色
 ```javascript
loadOml2d({primaryColor: 'pink'}) 
```
![PixPin_2024-05-04_09-43-21](https://github.com/oh-my-live2d/oh-my-live2d/assets/88269600/531d8d9a-603c-41f0-8cf7-cbe517e80967)


2. 如果你只想更改menu的颜色而不更改其它模块的颜色，可以传入 [itemStyle](https://oml2d.com/options/MenusOptions.html#itemstyle) 来修改样式
```javascript
loadOml2d({
    menus: () => {
      return {
        disable: false,
        itemStyle: {
          background: 'black',
          color: 'blue'
        }
      }
    }
  })
```
![PixPin_2024-05-04_09-42-02](https://github.com/oh-my-live2d/oh-my-live2d/assets/88269600/10d2f7d9-115c-41d8-8d16-293942c46858)

**Reply by Yansr82** [reply]
![image](https://github.com/oh-my-live2d/oh-my-live2d/assets/167418265/b6fc2bfe-3b00-4441-83b3-7c006a9fad18)
我意思是在這區塊建立一個"settings"可供用戶更改各自喜愛的主題色或是自定義菜單連結等等
只傳入用戶所選的主題色覆蓋掉預設的顏色

**Reply by KISGP** [reply]
具体我没有试过，应该是可行的。

比如可以先让用户输入设置，点击保存存储这些信息，然后刷新页面，最后在根据存储的信息初始化live2d。

**Reply by Yansr82** [reply]
那可能是我寫錯了
我再摸索看看，感謝！

**Reply by KISGP** [reply]
客气😚

**Comment by Yansr82** [comment]
你好，想請問若設置initialStatus: 'sleep' ，使其默認休息狀態的話
讀取模型完成後StatusBar也會收起，應該如何讓StatusBar配合initialStatus: 'sleep'保持開啟呢?


kisssssssss ***@***.***> 於 2024年5月5日 週日 上午1:17寫道：

**Reply by KISGP** [reply]
当设置 initialStatus: 'sleep' 时，StatusBar 隐藏是由于一些 bug 导致的，这个现在还没有修复。

这个 bug 的位置是 https://github.com/oh-my-live2d/oh-my-live2d/blob/main/packages/oh-my-live2d/src/modules/status-bar.ts 
![image](https://github.com/oh-my-live2d/oh-my-live2d/assets/88269600/47e268fa-a195-4474-8412-4963c17618df)
bug原因：当连续执行这个方法时，会无法清理上一个方法执行时注册的定时器。

**Comment by Yansr82** [comment]
*了解，所以目前還沒辦法解決這問題*
*謝謝回覆*
